pywps/app/Process.py

```diff
@@ -378,9 +381,13 @@ def _set_grass(self, wps_request):
 
             if new_loc_args:
                 dbase = self.workdir
-                location = 'pywps_location'
+                location = str()
+                while os.path.isdir(os.path.join(dbase, location)):
+                    location = 'pywps_loc_{}'.format(
+                        ''.join(random.sample(string.ascii_letters, 5)))
```

Comment:
```import pathlib
...
location = pathlib.Path(location).name
```

should do the job - let Python do, what it can do

Reply:
You mean something like:
```
import pathlib
location = pathlib.Path(dbase, 'pywps_loc_{}'.format(''.join(random.sample(string.ascii_letters, 5))))
while location.is_dir():
    location = pathlib.Path(dbase, 'pywps_loc_{}'.format(''.join(random.sample(string.ascii_letters, 5))))

location = pathlib.Path(location).name
```
?

Or do you mean just using the whole path in the loop (`os.path.join()` not in the condition, but in the body of the loop) and then derive the dir name through `pathlib`?

Because `pathlib` still doesn't give us a non-existing directory, we have to use a string as the parameter of `Path()`.

Reply:
nope, what I mean is: use standrad `tempfile` module which will make sure, there is no such "random" directory already available

Reply:
Then there would still be the same problem that `tempfile` creates a new directory. We just need a path to a non-existing directory to parse as an argument to GRASS functions used on the few lines lower in the code.